src/test/java/com/synopsys/integration/alert/web/certificates/CertificateActionsTestIT.java

```diff
@@ -29,6 +31,19 @@ public void createCertificateTest() throws Exception {
         assertEquals(certificateContent, savedCertificate.getCertificateContent());
     }
 
+    @Test
+    public void createCertificateIdTest() throws Exception {
+        String certificateContent = readCertificateContents();
+        try {
+            CertificateModel certificate = new CertificateModel("alias", certificateContent);
+            certificate.setId("badId");
+            certificateActions.createCertificate(certificate);
+            fail();
+        } catch (AlertDatabaseConstraintException ex) {
+            assertTrue(ex.getMessage().contains("id cannot be present"));
```

Comment:
I used to be in favor of checking the substrings in messages, but now I worry that it causes us to need to update the tests too often, which, in turn, causes the tests to be less valuable. I would be more comfortable having the fail at line 41 and nothing in the catch block (maybe an assertNotNull).

Reply:
I have changed to an assertNotNull for the message.